invoices/update.go

```diff
@@ -223,8 +235,18 @@ func updateLegacy(ctx *invoiceUpdateCtx,
 		return nil, ctx.failRes(ResultAmountTooLow), nil
 	}
 
-	// TODO(joostjager): Check invoice mpp required feature
-	// bit when feature becomes mandatory.
+	// If the invoice had the required feature bit set at this point, then
+	// if we're in this method it means that the remote party didn't supply
+	// the expected payload. However if this is a keysend payment, then
+	// we'll permit it to pass.
+	_, isKeySend := ctx.customRecords[record.KeySendType]
+	invoiceFeatures := inv.Terms.Features
+	paymentAddrRequired := invoiceFeatures.HasFeature(
```

Comment:
i think we need a new method here: `RequiresFeature`. `HasFeature` returns true if either are set.

Reply:
Added in the latest version! Also added a test for both the new method, as well as the positive path (for legacy invoices) in this area as well.